scripts/install.sh

```diff
@@ -0,0 +1,224 @@
+    cd ${SUBMODULE_DIR}/debug/robin-hood-hashing/build ; cmake --build ${SUBMODULE_DIR}/debug/robin-hood-hashing/build --target install -- -j${NUMBER_OF_PROCESSORS}
+
+    cd ${ROOT_DIR}
+}
+
+setup_validation_layers() {
+    update_submodules debug/Vulkan-ValidationLayers
+    
+    cd ${SUBMODULE_DIR}/debug/Vulkan-ValidationLayers ; git fetch --all --tags ; git checkout tags/v1.3.211
+
+    mkdirp ${SUBMODULE_DIR}/debug/Vulkan-ValidationLayers/build
+
+    cd ${SUBMODULE_DIR}/debug/Vulkan-ValidationLayers ; cmake \
+        -DVULKAN_HEADERS_INSTALL_DIR=${SUBMODULE_DIR}/Vulkan-Headers/build/install \
+        -DVULKAN_LOADER_INSTALL_DIR=${SUBMODULE_DIR}/Vulkan-Loader/build \
+        -DGLSLANG_INSTALL_DIR=${SUBMODULE_DIR}/glslang/build/install \
+        -DSPIRV_HEADERS_INSTALL_DIR=${SUBMODULE_DIR}/debug/SPIRV-Headers/build/install \
+        -DSPIRV_TOOLS_INSTALL_DIR=${SUBMODULE_DIR}/debug/SPIRV-Tools/build/install \
+        -DROBIN_HOOD_HASHING_INSTALL_DIR=${SUBMODULE_DIR}/debug/robin-hood-hashing/build/install \
+        -DCMAKE_BUILD_TYPE=Release \
+        -DCMAKE_INSTALL_PREFIX=./ \
+        -DBUILD_TESTS=OFF ..
+    
+    cd ${SUBMODULE_DIR}/debug/Vulkan-ValidationLayers/build ; cmake --build . --config Release --target install
+
+    mkdirp ${SUBMODULE_LIB_DIR}/explicit_layer.d
+
+    cp ${SUBMODULE_DIR}/debug/Vulkan-ValidationLayers/build/share/vulkan/explicit_layer.d/**.json ${SUBMODULE_LIB_DIR}/explicit_layer.d
+    cp ${SUBMODULE_DIR}/debug/Vulkan-ValidationLayers/build/layers/**.so ${SUBMODULE_LIB_DIR}/explicit_layer.d
+    cp ${SUBMODULE_LIB_DIR}/explicit_layer.d/**.so ${SUBMODULE_LIB_DIR}
+
+    cd ${ROOT_DIR}
+}
+
+setup_vulkan_validation_layers() {
+    setup_glslang
+    setup_robin_hood_hashing
+    setup_spirv_headers
+    setup_spirv_tools
+    setup_validation_layers
+}
+
+setup_volk
+setup_glfw
+
+if [[ -z "${VULKAN_SDK}" ]]; then
+    
+    echo "OS detected: ${OS}"
+    echo "# Environment variables for Vulkan." > .env
+
+    setup_vulkan_headers
+
+    echo "VULKAN_INCLUDE_DIR=${SUBMODULE_INCLUDE_DIR}/vulkan" >> .env
+
+    if [[ "${OS}" == "macOS" ]]; then 
+        
+        echo "VULKAN SDK has not been set. Setting up MoltenVK..."
+
+        setup_moltenVk
+
+        echo "MoltenVK successfully installed! Generating .env file..."
+
+        echo "VK_ICD_FILENAMES=${SUBMODULE_LIB_DIR}/icd.d/MoltenVK_icd.json" >> .env
+    fi 
+
+    if [[ "$OSTYPE" == "linux-gnu"* ]]; then
+
+        setup_vulkan_loader
+
+        if [[ "$1" == "DEBUG" ]]; then
+            
+            echo "Setting up Validation Layers..."
+
+            setup_vulkan_validation_layers
+
+            echo "VK_LAYER_PATH=${SUBMODULE_LIB_DIR}/explicit_layer.d" >> .env
+        fi
+    fi
+
+    echo "DYLD_LIBRARY_PATH=${SUBMODULE_LIB_DIR}" >> .env
+fi
```

Comment:
might want to issue a message informing the user that `VULKAN_SDK` was found, and is aborting

Reply:
Done.